### ISSUE TITLE
[Request] Basic ANSI/VT100 Control Sequence Colorization/Formatting Support

### ISSUE THREAD

**Issue by RichiCoder1** [highlightjs/highlight.js#3160]
**Is your request related to a specific problem you're having?**
Less of a problem and more of a hope.

**The solution you'd prefer / feature you'd like to see added...**
In essence, basic support for formatting or stripping out ANSI/VT100 codes for `console/shell-session`. Not full on terminal replay support, but basic support for translating color and formatting codes into corresponding markup and CSS and removing unsupported codes.

**Additional context...**
Basically to enable taking console output directly from colorized CLI output and embedding it in markdown. Evening more specifically, to enable colorized support for console output snippets in comments made by build tools without having to dive into terminal logs.

**Comment by joshgoebel** [comment]
Not sure I follow.  Please provide an example of input/output.

**Comment by RichiCoder1** [comment]
It may be non-trival to parse just because it's not symetrical like some grammars, but basically

```text
[31m
[1m[31mError: [0m[0m[1mNo configuration files[0m

[0mPlan requires configuration to be present. Planning without a configuration
would mark everything for destruction, which is normally not what is desired.
If you would like to destroy everything, run plan with the -destroy option.
Otherwise, create a Terraform configuration file (.tf file) and try again.
[0m[0m
```

to 

![image](https://user-images.githubusercontent.com/2391878/115640196-fc568c00-a2db-11eb-841b-53510edce5e2.png)
(span/css)


More context:
* All Codes: https://wiki.bash-hackers.org/scripting/terminalcodes
* Specifically Colorization and Formatting: https://stackoverflow.com/a/33206814
* Prior art stripping out ansi codes: https://www.npmjs.com/package/strip-ansi

**Comment by joshgoebel** [comment]
Sorry to crush your hopes, but this isn't what we do.  We highlight code - we do not modify it or otherwise attempt to "render" it.  If we supported ANSI (as a grammar) it would look just like the first capture but perhaps with the escape sequences highlighted so that they "popped", etc...

This is a great idea though.  Seems like the kind of thing that'd be pretty easy to do (at least for the FG/BG ANSI sequences - less so the move cursor, etc), but sounds like a separate library.

**Comment by RichiCoder1** [comment]
Fair enough :). Thank you!

**Comment by unphased** [comment]
I would love this for the convenience of being able to stick colorized terminal output into a code block and get it rendered as we would expect (by having something to interpret the escape sequences). It would be nice if highlightjs could do this, then it would work out of the box in something like reveal.js for presentations. 

Although it is true that ANSI escape sequences seems at first a completely different paradigm from what highlightjs does keeping content verbatim and applying styling...

**It seems possible if you just take the ansi escape sequences themselves, put those into spans, styled as `display:none` (as they are nonprintable anyhow and this is precisely what a terminal emulator would do).** Then styling the rest of the content in the appropriate spans would yield the correct result.

**Comment by joshgoebel** [comment]
Anyone wishing to write truly unique parsing engines is now free to do so via https://github.com/highlightjs/highlight.js/pull/3620.  Though the output must still be a valid HLJS token stream - ie all you can emit are named scopes and text, etc...

https://highlightjs.readthedocs.io/en/latest/mode-reference.html#emittokens
